prow/plugins/commands/commands.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+	Package commands provides a modular framework for defining command based Prow plugins.
+
+	Example label based commands:
```

Comment:
I'm worried this will drift and not be useful as we refactor. Maybe we can have an `example` package that ensures our documentation is compiling at the least?

Reply:
Good point. Once this framework is used in a few places I'll link to those packages as examples. I think that is just as useful as a dedicated `example` package, and doesn't involve up keeping unused code.

Reply:
Agree that an example.go is easier to consume than a 30 line comment